### ISSUE TITLE
Prometheus error: requests.exceptions.HTTPError: 422 Client Error

### ISSUE THREAD

**Issue by nizarh11** [scylladb/scylla-monitoring#1007]
*Installation details*
Scylla-Monitoring Version: branch-3.1
Scylla-Version: ami_id_db_scylla: ami-0eedbef0ebb289b75

I am getting the following error on Jenkins job "scylla-master-perf-regression-aws-test-latency" 
https://jenkins.scylladb.com/job/scylla-master/job/scylla-master-perf-regression-aws-test-latency/30/

**Error: requests.exceptions.HTTPError: 422 Client Error**

The same error happens 3-4 times but with a different query:

```
19:54:08  Exception happened during processing of request from ('172.30.0.90', 59698)
19:54:08  Traceback (most recent call last):
19:54:08    File "/usr/lib64/python3.6/socketserver.py", line 654, in process_request_thread
19:54:08      self.finish_request(request, client_address)
19:54:08    File "/usr/lib64/python3.6/socketserver.py", line 364, in finish_request
19:54:08      self.RequestHandlerClass(request, client_address, self)
19:54:08    File "/usr/lib64/python3.6/socketserver.py", line 724, in __init__
19:54:08      self.handle()
19:54:08    File "/usr/lib64/python3.6/http/server.py", line 418, in handle
19:54:08      self.handle_one_request()
19:54:08    File "/usr/lib64/python3.6/http/server.py", line 406, in handle_one_request
19:54:08      method()
19:54:08    File "/usr/local/lib/python3.6/site-packages/prometheus_client/exposition.py", line 102, in do_GET
19:54:08      self.wfile.write(output)
19:54:08    File "/usr/lib64/python3.6/socketserver.py", line 803, in write
19:54:08      self._sock.sendall(b)
19:54:08  BrokenPipeError: [Errno 32] Broken pipe
19:54:08  ----------------------------------------
19:54:08  /usr/local/lib/python3.6/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings

```

I tried running the following on the monitor itself and the query ran successfully:

```
http://54.85.253.78:9090/api/v1/query_range?query=histogram_quantile(0.99,%20sum(rate(scylla_storage_proxy_coordinator_read_latency_bucket%7B%7D[30s]))%20by%20(le))&start=1595777065&end=1595778874&step=5
```

The Monitor is still alive and can be accessed on "http://54.85.253.78:9090"


Logs:
```

+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                               Log links for testrun with test id 7c0fff58-1f4b-46ef-bf09-05cf1d532e37                                                                                                |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Date            | Log type    | Link                                                                                                                                                                                                                                 |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 20190101_010101 | prometheus  | https://cloudius-jenkins-test.s3.amazonaws.com/7c0fff58-1f4b-46ef-bf09-05cf1d532e37/prometheus_snapshot_20200726_150530.tar.gz                                                                                                       |
| 20200726_150319 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/7c0fff58-1f4b-46ef-bf09-05cf1d532e37/20200726_150319/grafana-screenshot-overview-20200726_150405-perf-regression-latency-perf-v8-monitor-node-7c0fff58-1.png                          |
| 20200726_150319 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/7c0fff58-1f4b-46ef-bf09-05cf1d532e37/20200726_150319/grafana-screenshot-scylla-per-server-metrics-nemesis-20200726_150328-perf-regression-latency-perf-v8-monitor-node-7c0fff58-1.png |
| 20200726_181708 | db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/7c0fff58-1f4b-46ef-bf09-05cf1d532e37/20200726_181708/db-cluster-7c0fff58.zip                                                                                                          |
| 20200726_181708 | loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/7c0fff58-1f4b-46ef-bf09-05cf1d532e37/20200726_181708/loader-set-7c0fff58.zip                                                                                                          |
| 20200726_181708 | monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/7c0fff58-1f4b-46ef-bf09-05cf1d532e37/20200726_181708/monitor-set-7c0fff58.zip                                                                                                         |
| 20200726_181708 | sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/7c0fff58-1f4b-46ef-bf09-05cf1d532e37/20200726_181708/sct-runner-7c0fff58.zip                                                                                                          |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

```

**Comment by amnonh** [comment]
@nizarh11 the monitoring stack does not use python, it seems like it's related to the testing infrastructure. Maybe @fruch would know.

Do you know what process those messages came from?

**Comment by fruch** [comment]
Issue moved to [scylladb/scylla-cluster-tests #2437](https://github.com/scylladb/scylla-cluster-tests/issues/2437) via [**ZenHub**](https://www.zenhub.com/)

**Comment by fruch** [comment]
@amnonh seems like @nizarh11 pointed the wrong backtrace:
```
[2020-07-26T18:17:01.323Z] Traceback (most recent call last):
[2020-07-26T18:17:01.323Z]   File "/sct/performance_regression_test.py", line 472, in test_latency
[2020-07-26T18:17:01.323Z]     self.run_read_workload()
[2020-07-26T18:17:01.323Z]   File "/sct/performance_regression_test.py", line 189, in run_read_workload
[2020-07-26T18:17:01.323Z]     self.update_test_details()
[2020-07-26T18:17:01.323Z]   File "/sct/sdcm/db_stats.py", line 570, in update_test_details
[2020-07-26T18:17:01.323Z]     update_data['results'] = self.get_prometheus_stats()
[2020-07-26T18:17:01.323Z]   File "/sct/sdcm/db_stats.py", line 507, in get_prometheus_stats
[2020-07-26T18:17:01.323Z]     prometheus_stats[stat] = self._calc_stats(ps_results=stat_calc_func(start_time=start, end_time=end))
[2020-07-26T18:17:01.324Z]   File "/sct/sdcm/db_stats.py", line 276, in get_latency_read_99
[2020-07-26T18:17:01.324Z]     return self.get_latency(start_time, end_time, latency_type="read")
[2020-07-26T18:17:01.324Z]   File "/sct/sdcm/db_stats.py", line 273, in get_latency
[2020-07-26T18:17:01.324Z]     return self._get_query_values(query, start_time, end_time)
[2020-07-26T18:17:01.324Z]   File "/sct/sdcm/db_stats.py", line 198, in _get_query_values
[2020-07-26T18:17:01.324Z]     results = self.query(query=query, start=start_time, end=end_time)
[2020-07-26T18:17:01.324Z]   File "/sct/sdcm/db_stats.py", line 182, in query
[2020-07-26T18:17:01.324Z]     result = self.request(url=_query)
[2020-07-26T18:17:01.324Z]   File "/sct/sdcm/utils/common.py", line 175, in inner
[2020-07-26T18:17:01.324Z]     return func(*args, **kwargs)
[2020-07-26T18:17:01.324Z]   File "/sct/sdcm/db_stats.py", line 146, in request
[2020-07-26T18:17:01.324Z]     response.raise_for_status()
[2020-07-26T18:17:01.324Z]   File "/usr/local/lib/python3.6/site-packages/requests/models.py", line 940, in raise_for_status
[2020-07-26T18:17:01.324Z]     raise HTTPError(http_error_msg, response=self)
[2020-07-26T18:17:01.324Z] requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: http://172.30.0.90:9090/api/v1/query_range?query=histogram_quantile(0.99,%20sum(rate(scylla_storage_proxy_coordinator_read_latency_bucket%7B%7D[30s]))%20by%20(le))&start=1595777065&end=1595778874&step=5
```

sounds like this issue:
https://github.com/prometheus/prometheus/issues/6516

**Comment by amnonh** [comment]
Thanks @fruch 
https://github.com/prometheus/prometheus/issues/6512 was solved in Prometheus 2.15.1
This is an old scylla monitoring release, scylla monitoring version 3.2 was already using Prometheus 2.15.2, please update to a later scylla monitoring.
Let me know if there is a need to backport the change
